### PR TITLE
Start implementing testing in cirrus

### PR DESCRIPTION
**Pull request by rhatdan**
Hook up regular tests to run on fedora/ubuntu natively.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by cevich** [comment]
@rhatdan this will help a lot with figuring out what steps are needed: #348

Otherwise the modify-rerun-CI loop will land you repeatedly waiting for the same package installs over and over and over.  Best figure out the steps manually first, then use CI in a PR to do a final check.

File: contrib/cirrus/setup.sh [hunk 1]
```diff
@@ -8,18 +8,30 @@ show_env_vars
...
     ubuntu-19)
         echo "Setting up $OS_RELEASE_ID $OS_RELEASE_VER"  # STUB: Add VM setup instructions here
+	apt-get -qq update
+	apt-get -qq install realpath $BuildRequires
+        install_fuse_overlay_from_git
         ;;
     ubuntu-18)
         echo "Setting up $OS_RELEASE_ID $OS_RELEASE_VER"  # STUB: Add VM setup instructions here
+	apt-get -qq update
+	apt-get -qq install realpath $BuildRequires
```
Comment (cevich) [inline contrib/cirrus/setup.sh]: apt-get is horrible to try and use under automation.  Other processes on the system will grab it's locks, and there's no provision to wait on them.  I'll scare up a retry/timeout loop I used in libpod for this...

Comment (cevich) [reply contrib/cirrus/setup.sh]: Okay, #349 should help with this.

**Comment by cevich** [comment]
@rhatdan I see things running, that's good.  I don't see a 10-minute timeout happening anywhere though, they look like regular test failures for one reason or another.  What am I missing?

**Comment by cevich** [comment]
Oh I see it now:

```
[+0791s] *** Test killed with quit: ran too long (10m0s).
```

That's coming from the test harness or the test itself, will have to dig deeper.

**Comment by podmanadm** [comment]
@cevich I think this is an actual bug in container/storage on f29.  I think Overlay is hanging in a deadlock.  For some reason this passes on F30.  I asked @nalind and @vrothberg To look at this, since I think we might have a deadlock causing it.

Not a cirrus problem.

**Comment by rhatdan** [comment]
podmanadm is my alterego

**Comment by cevich** [comment]
> I think this is an actual bug in container/storage on f29

@rhatdan I'm glad to hear CI is fulfilling it's purpose right out of the gate! :grinning:   That should be manually reproducible using that `hack/get_ci_vm.sh` by Nalin, but I don't think I gave Valentin access, I'll fix that now...

**Comment by rhatdan** [comment]
@cevich @nalind @TomSweeneyRedHat 
This test is only running overlay tests, right now.  Lets get this merged and I will continue to update this with enabling more tests.

**Comment by nalind** [comment]
Alright, looks fine as a step.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by cevich** [comment]
/lgtm

nice work.

**Pull request merged**